Yank/analyze.py

```diff
@@ -32,27 +35,1145 @@
+        generated_timeseries : 1-D iterable
+            timeseries which can be fed into get_decorrelation_time to get the decorrelation
+        """
+
+        raise NotImplementedError("This class has not implemented this function")
+
+    # Static methods
+    @staticmethod
+    def get_decorrelation_time(timeseries_to_analyze):
+        return timeseries.statisticalInefficiency(timeseries_to_analyze)
+
+    @staticmethod
+    def get_equilibration_data(timeseries_to_analyze):
+        [n_equilibration, g_t, n_effective_max] = timeseries.detectEquilibration(timeseries_to_analyze)
+        return n_equilibration, g_t, n_effective_max
+
+    @staticmethod
+    def remove_unequilibrated_data(data, number_equilibrated, axis):
+        """
+        Remove the number_equilibrated samples from a dataset by discarding number_equilibrated number of indices from
+        given axis
+
+        Parameters
+        ----------
+        data: np.array-like of any dimension length
+        number_equilibrated: int
+            Number of indices that will be removed from the given axis, i.e. axis will be shorter by number_equilibrated
+        axis: int
+            axis index along wich to remove samples from
+
+        Returns
+        -------
+        equilibrated_data: ndarray
+            Data with the number_equilibrated number of indices removed from the begining along axis
+
+        """
+        cast_data = np.asarray(data)
+        # Define the slice along an arbitrary dimension
+        slc = [slice(None)] * len(cast_data.shape)
+        # Set the dimension we are truncating
+        slc[axis] = slice(number_equilibrated, None)
+        # Slice
+        equilibrated_data = cast_data[slc]
+        return equilibrated_data
+
+    @staticmethod
+    def decorrelate_data(data, subsample_rate, axis):
+        """
+        Generate a decorrelated version of a given input data and subsample_rate along a single axis.
+
+        Parameters
+        ----------
+        data: np.array-like of any dimension length
+        subsample_rate : float or int
+            Rate at which to draw samples. A sample is considered decorrelated after every ceil(subsample_rate) of
+            indicies along data and the specified axis
+        axis: int
+            axis along which to apply the subsampling
+
+        Returns
+        -------
+        subsampled_data : ndarray of same number of dimensions as data
+            Data will be subsampled along the given axis
+
+        """
+        cast_data = np.asarray(data)
+        data_shape = cast_data.shape
+        # Since we already have g, we can just pass any appropriate shape to the subsample function
+        indices = timeseries.subsampleCorrelatedData(np.zeros(data_shape[axis]), g=subsample_rate)
+        subsampled_data = np.take(cast_data, indices, axis=axis)
+        return subsampled_data
+
+    # Private Class Methods
+    @abc.abstractmethod
+    def _prepare_mbar_input_data(self, *args, **kwargs):
```

Comment:
Thats probably a good idea. I think this `abstractmethod` was leftover from when this was a public function in my original design.